graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/Federation.java

```diff
@@ -76,4 +80,72 @@ private static void ensureQueryTypeExists(TypeDefinitionRegistry typeRegistry) {
             typeRegistry.add(ObjectTypeDefinition.newObjectTypeDefinition().name(queryName).build());
         }
     }
+
+    private static RuntimeWiring ensureFederationDirectiveDefinitionsExist(
+            TypeDefinitionRegistry typeRegistry,
+            RuntimeWiring runtimeWiring
+    ) {
+        // Add Federation directives if they don't exist.
+        FederationDirectives.allDefinitions
+                .stream()
+                // Really this sort is needed for tests, since SchemaPrinter doesn't sort directive
```

Comment:
Couldn't make tests use sets instead of lists and remove?

Reply:
In the tests, we compare SDL files against printed SDL. We could potentially split printed and expected SDL by `\n\n` into sets of (trimmed) strings, and test those sets for equality. Would that be good?

Reply:
Talked to Pierre on Slack, we decided to just use `LinkedHashSet` in `FederationDirectives` and insert in sorted order.